<!-- 
    ROS navigation stack with velocity smoother and safety (reactive) controller
-->
<launch>
  <!--include file="$(find navigation_sim_demo)/launch/include/velocity_smoother.launch.xml"/-->
  <!--include file="$(find navigation_sim_demo)/launch/include/safety_controller.launch.xml"/-->
  
  <arg name="odom_frame_id"   default="odom"/>
  <arg name="base_frame_id"   default="base_footprint"/>
  <arg name="global_frame_id" default="map"/>
  <arg name="odom_topic" default="odom" />
  <arg name="laser_topic" default="scan" />

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find wheel_leg)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find wheel_leg)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />   
    <rosparam file="$(find wheel_leg)/param/local_costmap_params.yaml" command="load" />   
    <rosparam file="$(find wheel_leg)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find wheel_leg)/param/dwa_local_planner_params.yaml" command="load" />
    <rosparam file="$(find wheel_leg)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find wheel_leg)/param/global_planner_params.yaml" command="load" />
    <rosparam file="$(find wheel_leg)/param/navfn_global_planner_params.yaml" command="load" />
    
    <!-- reset frame_id parameters using user input data -->
    <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
    <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="local_costmap/global_frame" value="$(arg odom_frame_id)"/>
    <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="DWAPlannerROS/global_frame_id" value="$(arg odom_frame_id)"/>

    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="scan" to="$(arg laser_topic)"/>
    <!-- <remap from="cmd_vel" to="/cmd_vel_mux/input/navi"/> -->

  </node>
</launch>
